### PR TITLE
Feedback style fixes

### PR DESCRIPTION
**Pull request by higs4281**
Finishing up styling and message interaction on new feedback modules to serve Owning a Home. 
This PR meshes with work @virginiacc is doing on the owning-a-home repo, which forwards non-wagtail feedback users to a wagtail feedback page. 

## Additions

- Contact_advisory richtext block to get long verbiage out of template.

## Changes

- Typos and style tweaks

## Testing

- Requires having owning-a-home running alongside cfgov-refresh locally. 
- Check out feedback forms on various pages:
  - http://localhost:8000/owning-a-home/
  - http://localhost:8000/owning-a-home/feedback/
  - http://localhost:8000/owning-a-home/help-us-improve/
  - http://localhost:8000/owning-a-home/process/prepare/

## Review

- @sebworks @chosak (block/migration changes) 

## Checklist

* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by sonnakim** [comment]
@sebworks @chosak would you guys mind reviewing this PR sometime today? Sorry to rush y'all, we have some dependencies!

**Comment by chosak** [comment]
@sonnakim sure, sorry for the delay.

**Comment by sonnakim** [comment]
@chosak no worries whatsoever - sorry to rush you!

File: cfgov/jinja2/v1/_includes/blocks/feedback.html [hunk 1]
```diff
@@ -11,11 +11,16 @@
...
       class="o-form
              block
              block__bg
              block__border
              block__padded-top">
+{% endif %}
+<div class="form-l">
     <input type="hidden" name="csrfmiddlewaretoken" value="{{ csrf_token }}">
     <input type="hidden" name="form_id" value="{{ form_id }}">
     <input type="hidden" name="referrer" value="{{ request.META.HTTP_REFERER }}">
```
Comment (chosak) [inline cfgov/jinja2/v1/_includes/blocks/feedback.html]: If you navigate to this page directly, and there's no referrer, then the form element looks like this:

image

To avoid this, can [this line](https://github.com/cfpb/cfgov-refresh/pull/2544/files#diff-0acfcc6db8ff7440bb3c2b303a8e3a4bL21) be changed to something like:

```html
<input type="hidden" name="referrer" value="{{ request.META.get('HTTP_REFERER') or '' }}">
```

File: cfgov/jinja2/v1/_includes/blocks/feedback.html [hunk 2]
```diff
@@ -58,14 +63,17 @@
             </div>
         </div>
     </fieldset>
+    {% else %}
+    <input type="hidden"
+           name="is_helpful"
+           value="{{ request.GET.get('is_helpful', '') }}">
```
Comment (chosak) [inline cfgov/jinja2/v1/_includes/blocks/feedback.html]: Instead of passing the request query parameter into the page like this, does it make sense to check the input (either through some logic here, or in a helper method) and force a value of `1` or `0`?

File: cfgov/v1/blocks.py [hunk 3]
```diff
@@ -109,7 +109,20 @@ class Feedback(AbstractFormBlock):
...
+    contact_advisory = blocks.RichTextBlock(
+        required=False,
+        default=('If you provide your contact information, '
+                 'we may contact you to clarify your question. '
+                 'This information is collected in accordance with the '
+                 '<a href="/owning-a-home/help-us-improve/'
+                 'paperwork-reduction-act-statement/">'
+                 'Paperwork Reduction Act Statement</a> and the '
+                 '<a href="/owning-a-home/help-us-improve/'
+                 'privacy-act-statement/">Privacy Act Statement</a>.')
```
Comment (chosak) [inline cfgov/v1/blocks.py]: It would be nice not to have to hardcode these links here. Since this is a RichTextBlock, can editors internal links to these actual Wagtail pages? Actually, can we just remove this default text entirely and just require content editors to add it? How many pages does this apply to?

File: cfgov/v1/handlers/blocks/feedback.py [hunk 4]
```diff
@@ -59,22 +59,37 @@ def get_response(self, form):
...
-                message = 'Something went wrong. Please try again.'
-                messages.error(self.request, message)
+            msg = 'Something went wrong. Please try again.'
 
-        return {'form': form}
+        if self.request.is_ajax():
+            return JsonResponse({'result': 'fail', 'message': msg})
+        else:
+            messages.error(self.request, msg)
+            return {'form': form}
```
Comment (chosak) [inline cfgov/v1/handlers/blocks/feedback.py]: If you come to the feedback form page with `?is_helpful=1` set, and then POST the form without filling out a message, this logic shows you the page again with an error message (e.g. "You must enter a comment") but then you lose the query string. Is there some way of setting the hidden field on the form directly here?

File: cfgov/v1/handlers/blocks/feedback.py [hunk 5]
```diff
@@ -59,22 +59,37 @@ def get_response(self, form):
...
+                )
+            else:
+                return JsonResponse(
+                    {'result': 'pass',
+                     'heading': '',
+                     'message': 'Thanks for your feedback!'}
+                )
         else:
             messages.success(self.request, message='Thanks for your feedback!')
             return HttpResponseRedirect(self.page.url)
```
Comment (chosak) [inline cfgov/v1/handlers/blocks/feedback.py]: If this used `request.path` instead then you wouldn't run into issues with content redirecting to www, etc.

File: cfgov/v1/handlers/blocks/feedback.py (hunk 4 above)
Comment (chosak) [inline cfgov/v1/handlers/blocks/feedback.py]: If you come to the feedback page with `?is_helpful=1` set, and then try to submit the form without entering a message, it re-renders the page but now you've lost the query string (because it POSTS back to the same URL). Is there some way of setting the field value directly on the form?

File: cfgov/v1/handlers/blocks/feedback.py [hunk 6]
```diff
@@ -49,7 +49,7 @@ def get_response(self, form):
         if form.is_valid():
             feedback = form.save(commit=False)
             is_helpful = self.request.POST.get('is_helpful', None)
-            if is_helpful is not None:
+            if is_helpful is not None and is_helpful != '':
```
Comment (chosak) [inline cfgov/v1/handlers/blocks/feedback.py]: You can cause a 500 by coming to the page with an invalid query string, e.g. `?is_helpful=text`.

What about using something like this instead:

```py
try:
    feedback.is_helpful = bool(int(is_helpful))
except ValueError:
    pass
```

File: cfgov/v1/handlers/blocks/feedback.py [hunk 7]
```diff
@@ -59,22 +59,37 @@ def get_response(self, form):
 
     def success(self):
         if self.request.is_ajax():
-            return JsonResponse({'result': 'pass'})
+            if get_feedback_type(self.block_value) == 'suggestion':
+                return JsonResponse(
+                    {'result': 'pass',
+                     'heading': 'Thank you!',
+                     'message': "Be sure to also sign up for our email list "
```
Comment (chosak) [inline cfgov/v1/handlers/blocks/feedback.py]: Would it be possible to have this success message be configurable on the block, instead of hardcoding it here? This makes the feedback block OAH-specific.

File: cfgov/jinja2/v1/_includes/blocks/feedback.html (hunk 2 above)
Comment (higs4281) [inline cfgov/jinja2/v1/_includes/blocks/feedback.html]: We want None as a possible value, because the model stores it that way (NullBooleanField) to handle feedback modules with no is_helpful question. So I removed the '' default so that it will pass None instead if the query string is not present.

File: cfgov/v1/blocks.py (hunk 3 above)
Comment (higs4281) [inline cfgov/v1/blocks.py]: I had included it as a convenience, to use the cleared language as a starting point that can be overridden at will in the admin, but we can remove the default text and let an editor enter it.

File: cfgov/v1/handlers/blocks/feedback.py (hunk 6 above)
Comment (higs4281) [inline cfgov/v1/handlers/blocks/feedback.py]: another good idea.

File: cfgov/v1/handlers/blocks/feedback.py (hunk 7 above)
Comment (higs4281) [inline cfgov/v1/handlers/blocks/feedback.py]: The radio-button suggestion format is indeed OAH-specific, and all other feedback forms should get the default success message. I think this could be part of a radio-button framework discussed earlier that is out of scope for this effort.

File: cfgov/v1/handlers/blocks/feedback.py (hunk 5 above)
Comment (higs4281) [inline cfgov/v1/handlers/blocks/feedback.py]: Grazie.

File: cfgov/v1/handlers/blocks/feedback.py (hunk 4 above)
Comment (higs4281) [inline cfgov/v1/handlers/blocks/feedback.py]: the best path may be to handle this error browser-side (which we've discussed with @virginiacc), because the full Django round-trip gets too convoluted -- both the query string and the referrer values pose chicken-and-egg problems. For now I think we can live with losing the `is_helpful` value if someone tries to submit an empty comment. `Is_helpful` is already recorded in analytics on the referring page, and it was considered a nice-to-have on the comment page.

**Comment by chosak** [comment]
:+1: Thanks for addressing my feedback @higs4281. I agree that a more configurable radio button approach would be a good future effort if/when additional feedback forms are needed elsewhere on the site.
